### PR TITLE
cargo-autoinherit: init at 0.1.3

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ca/cargo-autoinherit/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    owner = "mainmatter";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-BuZDCd4SwSYg5eKV61L3RpPVmq5NZDAOc9zOz5QiSNI=";
+  };
+
+  cargoHash = "sha256-9hhrVkC1xB2E/vatkiM4PIJyXq+0GDoqlgXZXc8WehU=";
+
+  meta = with lib; {
+    description = "(Auto)DRY for your Rust dependencies";
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-autoinherit/package.nix:17]:
```suggestion
    description = "Automatically DRY up your Rust dependencies";
```

make it less confusing

File: pkgs/by-name/ca/cargo-autoinherit/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = "https://github.com/mainmatter/cargo-autoinherit";
+    license = with licenses; [ asl20 /* OR */ mit ];
+    platforms = platforms.linux; # not tested on others
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-autoinherit/package.nix:20]:
```suggestion
    platforms = platforms.unix;
```

Would be fine. If you find it failing to build on CI (ofBorg/Hydra), just mark it as broken

File: pkgs/by-name/ca/cargo-autoinherit/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-autoinherit";
+  version = "0.1.4";
+
+  src = fetchFromGitHub {
+    owner = "mainmatter";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/ca/cargo-autoinherit/package.nix:9]:
```suggestion
    repo = "cargo-autoinherit";
```

File: pkgs/by-name/ca/cargo-autoinherit/package.nix (hunk 1 above)
Comment (matthiasbeyer) [reply pkgs/by-name/ca/cargo-autoinherit/package.nix:17]: Its the description of the repository :shrug:

File: pkgs/by-name/ca/cargo-autoinherit/package.nix (hunk 3 above)
Comment (matthiasbeyer) [reply pkgs/by-name/ca/cargo-autoinherit/package.nix:9]: Why?

File: pkgs/by-name/ca/cargo-autoinherit/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/ca/cargo-autoinherit/package.nix:17]: Description of the repository doesn't always make sense as well 🙃

File: pkgs/by-name/ca/cargo-autoinherit/package.nix (hunk 3 above)
Comment (Aleksanaa) [reply pkgs/by-name/ca/cargo-autoinherit/package.nix:9]: https://github.com/NixOS/nixpkgs/issues/277994

There is no necessary connection between the two, and it is useless to establish this relationship between them.

**Pull request merged**